### DISCUSSION TITLE
VRMAnimationLoaderPlugin で specVersion 未指定の vrma ファイルを読み込めるようにできないか？

### DISCUSSION THREAD

**Discussion by k1350** [pixiv/three-vrm#1563] [Help] [answered]
`@pixiv/three-vrm-animation` について質問があります。

下記の example で
https://pixiv.github.io/three-vrm/packages/three-vrm-animation/examples/dnd.html

下記の vrma ファイルは正常に読み込んで表示することができました。
https://booth.pm/ja/items/5512385

しかし幾つかの vrma ファイルが読み込めないことがわかりました。
たとえば下記が読み込めません。
https://booth.pm/ja/items/5527394

読み込むとコンソールには

```
VRMAnimationLoaderPlugin.ts:62 
 VRMAnimationLoaderPlugin: Unknown VRMC_vrm_animation spec version: undefined
```

という warning が出ていました。

試しに [VRoid Hub](https://hub.vroid.com/) で同じ vrma ファイルを読み込んだところ、VRoid Hub では読み込むことができました。

おそらく仕様上 specVersion は必須なので、 `VRMAnimationLoaderPlugin` ではバリデーションチェックに通らず読み込めていない。
一方 VRoid Hub では適当なバージョンとみなして読み込んでいるのだろうと推察しております。

 `VRMAnimationLoaderPlugin` の挙動は正しいと思いますが、現実として specVersion が無い vrma ファイルがそれなりに出回っているようでした。
できれば `VRMAnimationLoaderPlugin` でも読み込めてほしいのですが、難しいでしょうか？

**Comment by 0b5vr** [comment, marked as answer]
v0.120より前のバージョンのUniVRMにおいて、specVersionが含まれないVRMAが出力されることを確認しております。
https://github.com/vrm-c/UniVRM/pull/2251

> `VRMAnimationLoaderPlugin` の挙動は正しいと思いますが、現実として specVersion が無い vrma ファイルがそれなりに出回っているようでした。
> できれば `VRMAnimationLoaderPlugin` でも読み込めてほしいのですが、難しいでしょうか？

おっしゃるとおり現状、specVersionのないVRMAが多く出回っているようです。加えて、specVersionとして有効な値が1.0しかないことを鑑みて、specVersionが指定されていないものを読み込めるようにする変更を検討します。

**Reply by k1350** [reply]
ありがとうございます！

**Reply by 0b5vr** [reply]
`v3.3.2` において、 `specVersion` が含まれていないVRMAを1.0として解釈する変更を追加しました。

https://github.com/pixiv/three-vrm/releases/tag/v3.3.2

冒頭の投稿で参照されていたアニメーションも読めるようになりましたので、試してみてください。
